halo2_proofs/src/plonk/circuit.rs

```diff
@@ -392,13 +460,18 @@ impl Selector {
     pub fn is_simple(&self) -> bool {
         self.1
     }
+
+    /// Return expression from selector
+    pub fn expr<F: Field>(&self) -> Expression<F> {
+        Expression::Selector(*self)
+    }
 }
 
 /// Query of fixed column at a certain relative location
 #[derive(Copy, Clone, Debug)]
 pub struct FixedQuery {
     /// Query index
-    pub(crate) index: usize,
+    pub(crate) index: Option<usize>,
```

Comment:
To keep backward compatibility (to have make `VerifyingKey::hash_into` output the same initial transcript state), we'd need to implement the `Debug` manually to display it without `Some(...)`.